public/docs/ts/latest/guide/component-communication.jade

```diff
@@ -0,0 +1,707 @@
+  children in a trivial way like this:
+  
++makeExample('component-communication/ts/fragments/parent-reference-1.html')
+  
+:marked
+  Beside this very technique, there are other ways to pass the parent reference to children, with little tricks we can even use dependency injection
+  to inject the parent into the child component's constructor.
+  The concrete technique does not matter: **Angular discourages this model, because it can sidestep the change detection mechanism, and so it may break
+  the data flow model.**
+  
+  [Jump to details](#-3-child-to-parent-communication-with-direct-method-calling)
+
+  __#4: Duplex parent/child communication through a service object__
+  
+  Parent and child components share their responsibilities. In this sharing both component types have their own responsibilities including UI logic
+  and others. The parent and its children both can manage state information about the UI logic. Often, the majority of communication is about
+  managing the overall state information of this component fabric.
+  
+  In many cases the logic can be outsourced to a separate service that is the single source of truth, stores the state and provides operations to
+  access and alter the state. In such a scenario, parent and child components do not need to talk with each other directly, each of them 
+  communicates with the intermediary service:
+  
++makeExample('component-communication/ts/fragments/service-intro.ts', 'service')
+
+:marked
+  This service is injected into the parent and child components via Angular dependency injection (via component constructors):
+
++makeExample('component-communication/ts/fragments/service-intro.ts', 'components')
+
+:marked
+  Both component types can access the service instance injected into the constructor. The dependency injection mechanism takes care &mdash; 
+  unless we do not configure it in another way &mdash; that the parent and its child components use the same service instance.
+  We can design and implement the approprate service methods and properties, and access them from within the components. 
+  
+  [Jump to details](#-4-duplex-parent-child-communication-through-a-service-object)
+
+  __#5: Communication among unrelated components via a message bus__
+  
+  In many inter-component communication scenario, the communicating parties are not in a perent-child relation with each other. They can
+  be siblings, located in separate branches of the component hierarchy, or even be totally unrelated to each other. 
+  In such scenarios, components can communicate with each others directly or through intermediary object.
+  
+  In this chapter, we will create a message bus that is used as a common communication channel for sending and listening to messages:
+  
++makeExample('component-communication/ts/fragments/message-bus-intro.ts')
+
+:marked
+  We rely on Angular's dependency injection to allow the components to use the message bus.
```

Comment:
See my earlier note about this being structurally the same as the service example.